src/pages/TCY/hooks/useTcyClaimSaversNotification.tsx

```diff
@@ -0,0 +1,68 @@
+import type { ToastId } from '@chakra-ui/react'
+import { useToast } from '@chakra-ui/react'
+import { useEffect, useRef } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useNavigate } from 'react-router-dom'
+
+import { TcyClaimSaversNotification } from '@/components/Layout/Header/ActionCenter/components/Notifications/TcyClaimSaversNotification'
+import { useWallet } from '@/hooks/useWallet/useWallet'
+import { ActionStatus } from '@/state/slices/actionSlice/types'
+import { preferences } from '@/state/slices/preferencesSlice/preferencesSlice'
+import { selectTcyClaimActionsByWallet } from '@/state/slices/selectors'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+export const useTcyClaimSaversNotification = () => {
+  const toast = useToast()
+  const navigate = useNavigate()
+  const translate = useTranslate()
+  const dispatch = useAppDispatch()
+
+  const hasSeenTcyClaimAlert = useAppSelector(preferences.selectors.selectHasSeenTcyClaimAlert)
+  const toastIdRef = useRef<ToastId | undefined>(undefined)
+
+  const tcyClaimActions = useAppSelector(selectTcyClaimActionsByWallet)
+  const {
+    state: { isConnected, walletInfo },
+  } = useWallet()
+
+  useEffect(() => {
+    const hasClaimable = tcyClaimActions.some(claim => claim.status === ActionStatus.ClaimAvailable)
+
+    // Everything already claimed or swapped to wallet with no claims (close toast)
+    if (!hasClaimable && toastIdRef.current) {
+      toast.close(toastIdRef.current)
+    }
```

Comment:
Same q as before @premiumjibles - do we need all this logic? see e.g 

https://github.com/shapeshift/web/blob/7d7af21c0478a33bd7ce88ea5d19290ca4fe1789/src/components/Layout/Header/ActionCenter/components/Notifications/SwapNotification.tsx#L25

There is no explicit close logic needed per se - more like, if the `byWallet` selector for that notification didn't return results, return null.

Also feels weird to have this logic in a hook, when really it could be in notification logic, WDYT?